Automattic/pocket-casts-android
# Bookmarks - Add bookmark from headset hook

## Description

This adds bookmarks using the [headset hook](https://developer.android.com/reference/android/view/KeyEvent#KEYCODE_HEADSETHOOK) button.

The headset hook button is already used to skip forward on double tap. The functionality is extended to add a bookmark based on how it is configured in "Headphone controls" settings.

## Testing Instructions
1. Install debug build
2. Launch the app 
3. Login to a paid account
4. Go `Profile` tab -> `Settings` -> `Headphone controls `
5. Tap on "Double tap headset hook button"
6. Select "Add bookmark" option
7. Go to "Discover" tab and play any episode
8. Connect headphone to the device 
9. With the app open, double tap headset hook button
    [Or you can run command `adb shell input keyevent 79` quickly twice to emulate the action]
10. ✅ Notice that "Add bookmark "screen is displayed
11. Dismiss the screen
12. Background the app
13. Again trigger the double tap headset hook button action
14. ✅ Notice that a notification is shown that the bookmark is added and sound is played
15. Tap on the notification 
16. App opens and navigates to the Player - Bookmarks screen 
17. Dismiss the screen and background the app again
18. Again trigger the double tap headset hook button action
19. ✅ Notice that a notification is shown that the bookmark is added and sound is played
20. Tap on the "Change title" notification action
21. ✅ Notice that "Change title" bookmark screen is shown
22. Update the title
23. ✅ Notice that snackbar appears with an action to view bookmark
24. Tap on view bookmark
25. ✅  Notice that  the app navigates to the Player - Bookmarks screen 

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/95bea484-055c-4a01-849e-b1453f76d0f8


Notification on Android API Level 23 (Action icon not shown on [versions above)](https://android-developers.googleblog.com/2016/06/notifications-in-android-n.html)

<img width="264" alt="notification" src="https://github.com/Automattic/pocket-casts-android/assets/1405144/43fe9f79-d9a7-4366-942c-a5d806deacea">



## Checklist
 
#### I have tested any UI changes...
